src/varity/chain.clj

```diff
@@ -122,3 +133,18 @@
   [chr pos chain-idx]
   (->> (chain-idx (normalize-chr chr))
        (keep (partial in-block? pos))))
+
+(defn search-overlap-chains
+  "Calculats a list of chains that overlap the given interval."
+  [chr start end chain-idx]
+  (->> (chain-idx (normalize-chr chr))
+       (filter #(and (<= (get-in % [:header :t-start]) start)
+                     (<= end (get-in % [:header :t-end]))))))
+
+(defn search-overlap-blocks
+  "Calculats a list of blocks that overlap the given interval."
+  [start end ^Sorted blocks-idx]
+  (->> (subseq blocks-idx <= end)
+       (take-while (fn [[block-start block]]
+                     (<= block-start end)))
```

Comment:
Here, `take-while` doesn't seem to do anything meaningful because `(subseq blocks-idx <= end)` ensures that all the returned entries have a key that is less than or equal to `end`. Probably, you need to do some more using `start` to cut down the head of the subsequence, right?